src/imageResponse.js

```diff
@@ -127,12 +127,22 @@ export default {
     const clientStartTime = new Date();
     const browserImage = await image.magic(imagePath(params.name), params, response);
     browserImage.stream(params.type, (err, stdout) => {
+      if(err) {
+        response.status(500).end();
+        log('error', `Error occurred while creating live image: ${err}`);
+        return;
+      }
       const r = stdout.pipe(response);
       r.on('finish', () => {
         log('info', `Creating live image took ${new Date() - clientStartTime}ms: ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)})`);  //eslint-disable-line max-len
         // This is to close the response while a background job will continue to process
         response.end();
       });
+      r.on('error', (error) => {
+        log('error', `The live image stream errorred with ${error}`);
+        response.end();
+        r.end();
```

Comment:
`r === response`, you're calling `end` on the same stream twice. (likewise, line 139 should also be removed, since the response is already closed at that point).

Before ending the response stream you should first `unpipe` it from `stdout`. If not, then `stdout` will still be able to send data/errors to response, which will obviously fail. 

@AlexKolpa is also right about this also triggering `finish`. It *should* trigger finish, but maybe you can keep add a flag to keep track if the response ended successful.